integration-tests/src/test_loop/tests/fix_chunk_producer_stake_threshold.rs

```diff
@@ -0,0 +1,145 @@
+            validators_spec,
+        },
+        |genesis_builder| genesis_builder,
+        |epoch_config_builder| epoch_config_builder,
+    );
+
+    let TestLoopEnv { mut test_loop, datas: node_data, tempdir } = TestLoopBuilder::new()
+        .genesis(genesis)
+        .epoch_config_store(epoch_config_store.clone())
+        .clients(clients)
+        .build();
+
+    let sender = node_data[0].client_sender.clone();
+    let handle = sender.actor_handle();
+    let client = &test_loop.data.get(&handle).client;
+
+    // premise checks
+    let epoch_id = client
+        .epoch_manager
+        .get_epoch_id_from_prev_block(&client.chain.head().unwrap().last_block_hash)
+        .unwrap();
+    let protocol_version = client.epoch_manager.get_epoch_protocol_version(&epoch_id).unwrap();
+    let validators = get_epoch_all_validators(client);
+    assert!(
+        protocol_version < ProtocolFeature::FixChunkProducerStakingThreshold.protocol_version()
+    );
+    assert_eq!(
+        epoch_config_store.get_config(protocol_version).shard_layout.num_shards(),
+        num_shards
+    );
+    assert_eq!(
+        validators,
+        vec![String::from("test0"), String::from("test1"), String::from("test3")]
+    );
+
+    test_loop.run_until(
+        |test_loop_data: &mut TestLoopData| {
+            let client = &test_loop_data.get(&handle).client;
+            let head = client.chain.head().unwrap();
+            let epoch_height = client
+                .epoch_manager
+                .get_epoch_height_from_prev_block(&head.prev_block_hash)
+                .unwrap();
+            // ensure loop is exited because of condition is met
+            assert!(epoch_height < 3);
+
+            let epoch_id = client
+                .epoch_manager
+                .get_epoch_id_from_prev_block(&client.chain.head().unwrap().last_block_hash)
+                .unwrap();
+            let protocol_version =
+                client.epoch_manager.get_epoch_protocol_version(&epoch_id).unwrap();
+            // exits when protocol version catches up the fix
+            protocol_version >= ProtocolFeature::FixChunkProducerStakingThreshold.protocol_version()
+        },
+        Duration::seconds(4 * epoch_length as i64),
+    );
+
+    test_loop.run_until(
+        |test_loop_data: &mut TestLoopData| {
+            let client = &test_loop_data.get(&handle).client;
+            let head = client.chain.head().unwrap();
+            let epoch_height = client
+                .epoch_manager
+                .get_epoch_height_from_prev_block(&head.prev_block_hash)
+                .unwrap();
+            // ensure loop is exited because of condition is met
+            assert!(epoch_height < 5);
+
+            // after the fix, threshold is raised to approximately 6x of the previous in this case
+            // (reason of "approximately 6x": ./fix_stake_threshold.rs, but it's irrelevant here)
+            // so test3's proposal will not be approved
```

Comment:
Let's just say that 6 corresponds to the number of shards, especially if fix_stake_threshold is irrelevant.